### DISCUSSION TITLE
docs: callout potentially breaking typescript version updates

### DISCUSSION THREAD

**Discussion by abhijeetviswa** [mikro-orm/mikro-orm#5613] [General]
Came to realize that the Typescript version was bumped in this commit: 75e43411dc7d02fa259299e84bb0c32498810d17 which also realized the "shim" `NoInfer` utility type for the Lib type. 

Looking at the Git tags, seems like this was first part of the v6.1.9. This is effectively a breaking change if consumers weren't already on Typescript 5.4 or have had the compiler flag `skipLibCheck` enabled.

**Comment by B4nan** [comment]
If you think it's breaking, provide a reproduction to prove that, AFAIK it still works just fine with older TS versions. Bumping TS versions is not breaking change, and the `NoInfer` helper [is still there](https://github.com/mikro-orm/mikro-orm/blob/master/packages/core/src/typings.ts#L48).

**Reply by abhijeetviswa** [reply]
The typings.ts variant you referred is not being imported in this file: `https://github.com/mikro-orm/mikro-orm/blob/master/packages/core/src/EntityManager.ts`. However, `NoInfer` is being used within that file. I'm not sure if the helper defined by MikrOrm can be used globally, without importing it. 

Here are some sample errors I'm facing:

```
node_modules/@mikro-orm/core/EntityManager.d.ts:293:92 - error TS2304: Cannot find name 'NoInfer'.

293     nativeDelete<Entity extends object>(entityName: EntityName<Entity>, where: FilterQuery<NoInfer<Entity>>, options?: DeleteOptions<Entity>): Promise<number>;
                                                                                               ~~~~~~~

node_modules/@mikro-orm/core/EntityManager.d.ts:365:115 - error TS2304: Cannot find name 'NoInfer'.

365     count<Entity extends object, Hint extends string = never>(entityName: EntityName<Entity>, where?: FilterQuery<NoInfer<Entity>>, options?: CountOptions<Entity, Hint>): Promise<number>;
```

These are the versions as per package-lock.json I'm using:

```
@mikro-orm/core = ^6.2.3
@mikro-orm/postgresql =  ^6.2.3
@mikro-orm/reflection = ^6.2.3
typescript = ~5.3.2
```

I'll try providing a reproducible git repository, but it may take some time. Hopefully the above details is enough in the meantime. Do let me know if I'm missing anything.

Thanks

**Reply by B4nan** [reply]
> The typings.ts variant you referred is not being imported in this file: https://github.com/mikro-orm/mikro-orm/blob/master/packages/core/src/EntityManager.ts. However, NoInfer is being used within that file. I'm not sure if the helper defined by MikrOrm can be used globally, without importing it.

Well, but this is a bug to fix, not a random breaking change to document, good catch!

**Reply by abhijeetviswa** [reply]
I understand now it's a bug. 

Appreciate the help and prompt response. Do you want me to raise this as an issue or will you be fixing this directly?

**Reply by B4nan** [reply]
I'll fix it later today, no need for another issue.

**Reply by B4nan** [reply]
c38b36609a5264a2b2a49a87a89091003fcf6f42
